integ-test/src/test/java/org/opensearch/sql/bwc/SQLBackwardsCompatibilityIT.java

```diff
@@ -93,21 +93,33 @@ public void testBackwardsCompatibility() throws Exception {
       List<Map<String, Object>> plugins = (List<Map<String, Object>>) response.get("plugins");
       Set<Object> pluginNames =
           plugins.stream().map(map -> map.get("name")).collect(Collectors.toSet());
+      String version = (String) response.get("version");
+
+      boolean isBackwardsIncompatibleVersion = version.startsWith("2.");
```

Comment:
would we test 2.x using main code? I thought we would remove 2.x from build.gradle so this won't happen, since they are not backward compatible

Reply:
> would we test 2.x using main code? I thought we would remove 2.x from build.gradle so this won't happen, since they are not backward compatible

I think the majority of the features are still backward compatible, besides those planned to deparcate in 3.0. Maybe discuss the bwc version of main branch on thread https://github.com/opensearch-project/sql/issues/1880.